composer-require-checker.json

```diff
@@ -0,0 +1,13 @@
+{
+    "php-core-extensions" : [
```

Comment:
The config is there in order to whitelist `iconv` and `mbstring` - they are suggested and tested to use one or the other _(Or throw)_. According to the docs for require-checker, modifying any config section requires that you repeat the defaults because only 1 config file is evaluated. I figured adding these as core extensions was cleaner than whitelisting all the symbols (Including the defaults too).

Reply:
Could we perhaps drop one of the two, and make the other always required? I don't know of PHP systems that don't have mbstring :thinking:

Reply:
Would that not require a new major? It's pretty trivial to just go with mbstring here. Might as well drop 7.3 too and update all the dev dependencies. Psalm is well out of date so there will probably be a load of fixes there too. Also, I can't for the life of me get infection running on this component.

Reply:
No, IMO sufficient to have a new minor release with the requirement here.

Reply:
> Also, I can't for the life of me get infection running on this component.

Any detail here? It is probably going to be a mess due to the huge amount of AST, I suppose?

Reply:
Nope, no detail. Maybe I'm doing the exclusions wrong for `test` and `vendor`, but it just borks on a failed assertion in Webmozart\Assert which is not unused in this component.

https://github.com/gsteel/laminas-escaper/runs/5463587033?check_suite_focus=true

Reply:
I think trying it manually via `phpdbg -qrr vendor/bin/infection -vvv` would show a stack trace here - that crash seems to be internal to infection itself.

Reply:
Sussed it. The weird codepoints generated by the data providers were causing code coverage XML hell… I modified the tests to accept the code point as an int, instead of the character string so coverage has valid XML and infection can do its thing…